### ISSUE TITLE
[Bug] [a11y] Hyper-linked texts aren't read

### ISSUE THREAD

**Issue by softvision-miralobontiu** [mozilla-mobile/fenix#4805]
## Steps to reproduce
1. Turn Talkback or Voice assistant on.
2. Open a page that contains hyper-linked texts.

### Expected behavior
The voice assistant should read the entire text.

### Actual behavior
The voice assistant doesn't read hyper-linked texts.

### Device information
* Android device: Xiaomi Mi8 Lite (Android 9)
* Fenix version: RC 1.3.0-rc.3

**Comment by cpeterso** [comment]
@softvision-miralobontiu - Does Talkback and Voice Assistant work correct in Fennec on the same device or in Fenix on other devices?

@eeejay - Do you have any theories why neither Talkback nor Voice Assistant is working in Fenix/GeckoView?

**Comment by softvision-miralobontiu** [comment]
@cpeterso it is also reproducible on the latest Firefox Nightly 68.2a1 (2019-09-08) from Google Play.
This issue also reproduces on Google Pixel 3 (Android 9).

**Comment by eeejay** [comment]
@softvision-miralobontiu, can you please provide a URL and the specific steps you took? Did you explore by touch to the link? Did you swipe to it?

**Comment by softvision-miralobontiu** [comment]
@eeejay I was on Wikipedia main page - English section - the article from yesterday.

**Comment by eeejay** [comment]
This works for me, is there something you are doing different?
![screencast](https://user-images.githubusercontent.com/5764/65267744-80c2a680-daca-11e9-9013-ed8ff06f5038.gif)

**Comment by softvision-miralobontiu** [comment]
Tested again on Xiaomi Mi 8 Lite (Android 9), and Google Pixel 3 (Android 9) on the latest Firefox Preview Nightly 190919 @6:19 (#12621820).
From the screenshot below, the voice assistant only read "to hang outside the Sutton...by", skipped National Trust from the beginning of the selected text and Nobel from the end of the selected text. 
I've attached a logcat, I don't know if that helps.
Still reproduces on Firefox 68.1. and Firefox Beta 68.2b2.

[logcat.txt](https://github.com/mozilla-mobile/fenix/files/3634927/logcat.txt)

![Screenshot_2019-09-20-11-33-23-646_org mozilla fenix nightly](https://user-images.githubusercontent.com/40694158/65312175-a4edb880-db9a-11e9-92f8-b28e54a2a79c.png)

**Comment by eeejay** [comment]
That is the expected behavior. If you swipe right again it will say "Nobel, link". It is designed that way so that the user can distinguish between differently formatted or interactive ranges of text.

**Comment by cpeterso** [comment]
> That is the expected behavior. If you swipe right again it will say "Nobel, link". It is designed that way so that the user can distinguish between differently formatted or interactive ranges of text.

Since this appears to be expected behavior, I will remove this issue's `needs:gv` label. Someone on the Fenix team can decide whether they want to accept that resolution (and close this bug) or suggest different behavior.

**Comment by mcarare** [comment]
IMO @eeejay has a good point. Also, this seems to be the default behaviour in both Chrome and Fennec. 
I am going to close this. If this should prove to be a problem for users we can revisit finding a solution.
